crates/ruff_linter/src/checkers/ast/analyze/bindings.rs

```diff
@@ -87,5 +88,10 @@ pub(crate) fn bindings(checker: &mut Checker) {
                 checker.diagnostics.push(diagnostic);
             }
         }
+        if checker.enabled(Rule::AssignmentInAssert) {
+            if let Some(diagnostic) = ruff::rules::assignment_in_assert(checker, binding) {
+                checker.diagnostics.push(diagnostic);
+            }
+        }
```

Comment:
Should we just update the `diagnostics` vector inside the rule logic to avoid doing this here? We usually do that in other rules.

Reply:
We do that for most rules in `expressions.rs` and `statement.rs`, but we don't do that for any rules in `bindings.rs` for some reason. I'm happy to look separately into whether we should change the structure of these rules to make them more consistent with the majority of other AST-based rules, but for now I think local consistency within this file probably wins.

Reply:
Oh, that's fine then, no need to spend any more time here.